### PR TITLE
CLOUDP-277319: Move RBAC from template to a separate file

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: charts/atlas-operator/templates/cluster-roles.yaml [hunk 1]
```diff
@@ -9,7 +9,10 @@ metadata:
   labels:
   {{- include "mongodb-atlas-operator.labels" $ | nindent 4 }}
 rules:
-{{ template "mongodb-atlas-operator.rbacRules" $ | toYaml | indent 2 }}
+{{- range (.Files.Lines "rbac.yaml") }}
+  {{ . -}}
+{{- end }}
+{{/*{{ template "mongodb-atlas-operator.rbacRules" $ | toYaml | indent 2 }}*/}}
```
Comment (josvazg) [inline charts/atlas-operator/templates/cluster-roles.yaml:15]: Is this line a comment? can we remove it?

File: charts/atlas-operator/templates/roles.yaml [hunk 2]
```diff
@@ -14,7 +14,9 @@ metadata:
   labels:
   {{- include "mongodb-atlas-operator.labels" $ | nindent 4 }}
 rules:
-{{ template "mongodb-atlas-operator.rbacRules" $ | toYaml | indent 2 }}
+{{- range (.Files.Lines "rbac.yaml") }}
```
Comment (josvazg) [inline charts/atlas-operator/templates/roles.yaml:17]: Why do we need to have the RBAC definitions repeated 3 times?

File: charts/atlas-operator/templates/cluster-roles.yaml (hunk 1 above)
Comment (igor-karpukhin) [reply charts/atlas-operator/templates/cluster-roles.yaml]: Thanks! Fixed

File: charts/atlas-operator/templates/roles.yaml (hunk 2 above)
Comment (igor-karpukhin) [reply charts/atlas-operator/templates/roles.yaml:17]: The reason these included twice, is that there are two sections for Roles: one for when AKO works one of managed namespaces, and when AKO is not in the managed namespaces list.

Comment (josvazg) [reply charts/atlas-operator/templates/roles.yaml:17]: It is included 3 times total. One in cluster roles and two here.

I see the ifs now for watched namespaces and not watched. Thanks for explaining.

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**